src/ast/sym.cc

```diff
@@ -149,15 +149,7 @@ namespace sym
 
       case "function"_:
       {
-        // A missing function name is sugar for "apply"
-        if (ast->nodes[0]->nodes.size() == 0)
-        {
-          ast->nodes[0]->nodes.push_back(
-            ast::from(ast->nodes[0], "id", "apply"));
-        }
-
-        auto node = ast->nodes[0]->nodes[0];
-        add_symbol(node->token, ast, err);
+        add_symbol(ast->nodes[0]->nodes[0]->token, ast, err);
```

Comment:
Why is apply sugar being removed?

Reply:
I think this node is only for function declarations. I think apply sugar only applies to function application, where it is parsed as:
```
+ term
    - prefix (foo)
    + tuple
```

Reply:
This changes an empty function name at declaration to instead be `apply`. This allows `apply` to be used to reference the function. There will be `apply` sugar at invocation time as well, but it's not in yet.

Reply:
Ok, I missed that the grammar allowed an empty function name. A test has been added for this sugar.